src/components/component/toolbar.tsx

```diff
@@ -0,0 +1,201 @@
+import { Button, CircularProgress, Typography } from '@equinor/eds-core-react';
+import * as PropTypes from 'prop-types';
+import { Component, MouseEvent } from 'react';
+import { connect } from 'react-redux';
+import { Dispatch } from 'redux';
+
+import {
+  componentRestartState,
+  componentStartState,
+  componentStopState,
+} from '../../state/component';
+import { actions as componentActions } from '../../state/component/action-creators';
+import { ComponentStatus } from '../../models/component-status';
+import { RequestState } from '../../state/state-utils/request-states';
+import { RootState } from '../../init/store';
+import {
+  ComponentModel,
+  ComponentModelValidationMap,
+} from '../../models/component';
+
+interface ToolbarData {
+  appName: string;
+  envName: string;
+  component?: ComponentModel;
+  startEnabled?: boolean;
+  stopEnabled?: boolean;
+}
+
+interface ToolbarAction {
+  startComponent: (
+    appName: string,
+    envName: string,
+    componentName: string
+  ) => void;
+  stopComponent: (
+    appName: string,
+    envName: string,
+    componentName: string
+  ) => void;
+  restartComponent: (
+    appName: string,
+    envName: string,
+    componentName: string
+  ) => void;
+}
+
+interface ToolbarState {
+  startRequestStatus: RequestState;
+  startRequestMessage: string;
+  stopRequestStatus: RequestState;
+  stopRequestMessage: string;
+  restartRequestStatus: RequestState;
+  restartRequestMessage: string;
+}
+
+interface ToolbarProps extends ToolbarData, ToolbarAction, ToolbarState {}
+
+export class Toolbar extends Component<ToolbarProps> {
+  static readonly propTypes: PropTypes.ValidationMap<ToolbarProps> = {
+    appName: PropTypes.string.isRequired,
+    envName: PropTypes.string.isRequired,
+    component: PropTypes.shape(
+      ComponentModelValidationMap
+    ) as PropTypes.Validator<ComponentModel>,
+    startEnabled: PropTypes.bool,
+    stopEnabled: PropTypes.bool,
+    startComponent: PropTypes.func.isRequired,
+    stopComponent: PropTypes.func.isRequired,
+    restartComponent: PropTypes.func.isRequired,
+    startRequestStatus: PropTypes.oneOf(Object.values(RequestState)),
+    startRequestMessage: PropTypes.string,
+    stopRequestStatus: PropTypes.oneOf(Object.values(RequestState)),
+    stopRequestMessage: PropTypes.string,
+    restartRequestStatus: PropTypes.oneOf(Object.values(RequestState)),
+    restartRequestMessage: PropTypes.string,
```

Comment:
According to the `ToolbarState` interface these props are supposed to be required
```suggestion
    startRequestStatus: PropTypes.oneOf(Object.values(RequestState)).isRequired,
    startRequestMessage: PropTypes.string.isRequired,
    stopRequestStatus: PropTypes.oneOf(Object.values(RequestState)).isRequired,
    stopRequestMessage: PropTypes.string.isRequired,
    restartRequestStatus: PropTypes.oneOf(Object.values(RequestState))
      .isRequired,
    restartRequestMessage: PropTypes.string.isRequired,
```